### ISSUE TITLE
ERROR: 400 Bad Request on /api/developer-scans/xxx?offset=0&limit=100

### ISSUE THREAD

**Issue by guellaeq** [blackduck-inc/black-duck-security-scan#28]
I am using the action in a pull request scan setup :

```yml      
      - name: Black Duck Scan
        if: ${{ github.event_name == 'pull_request' }}
        uses: blackduck-inc/black-duck-security-scan@v2.0.0
        with:
          blackducksca_url: ${{ vars.BLACKDUCK_URL }}
          blackducksca_token: ${{ secrets.BLACKDUCK_TOKEN }}
          blackducksca_scan_full: false
          blackducksca_waitForScan: true 
          blackducksca_prComment_enabled: true
          github_token: ${{ secrets.GITHUB_TOKEN }}
```
          
In the logs, I can see the rapid scan being uploaded to the temp server storage for analysis :
`INFO: --- Uploaded Rapid Scan: https://blackduck.***.com/api/developer-scans/***`

Later in the logs, detect is unable to fetch from this same url above to complete the run : 

```
INFO: --- ======== Detect Issues ========
INFO: --- 
INFO: --- EXCEPTIONS:
INFO: --- 	Rapid Wait
INFO: --- 		There was a problem trying to GET https://blackduck.***.com/api/developer-scans/***?offset=0&limit=100, response was 400 Bad Request.
```
This issue has already been reported before when using the deprecated synopsys-action : https://github.com/synopsys-sig/synopsys-action/issues/197

Additional info : Detect Version: 10.0.0

**Comment by guellaeq** [comment]
This error seems to come from the default detect configuration when using the action. 
This setting : "_detect.blackduck.rapid.compare.mode = BOM_COMPARE_STRICT [cmd]_" is not the same as detect default in the documentation : https://documentation.blackduck.com/bundle/detect/page/properties/configuration/rapid-scan.html#rapid-compare-mode-advanced
This has an impact on the rapid scan which expects a project name and project version to be able to attach and compare the BOM of your PR with the BOM of your project fully scanned in the server. 
This behavior is not documented anywhere.

In a nutshell, to fix this issue, I had to configure the project name and version to be retrieved from the GitHub context :
```yaml
env: 
  DETECT_PROJECT_NAME: ${{ github.repository_name }}
  DETECT_PROJECT_VERSION_NAME: ${{ github.event.pull_request.base.ref }}
```
It only works if the version exists and the _github.event.pull_request.base.ref_ branch has been fully scanned

**Comment by terryNDiko** [comment]
I had the same issue, to fix it, I had to add --detect.blackduck.rapid.compare.mode=ALL to the detect_args input
`      - name: Black Duck SCA Rapid Scan
        if: ${{ github.event_name == 'pull_request' }}
        uses: blackduck-inc/black-duck-security-scan@v2.0.0
        env:
          DETECT_PROJECT_NAME: ${{ github.event.repository.name }}
          DETECT_PROJECT_VERSION_NAME: ${{ github.event.pull_request.base.ref }}

        with:
            github_token: ${{ secrets.GITHUB_TOKEN }}
            blackducksca_url: ${{ secrets.BLACKDUCK_URL }} 
            blackducksca_token: ${{ secrets.BLACKDUCK_API_KEY }}
            detect_search_depth: "1"
            blackducksca_scan_full: "false"
            blackducksca_prComment_enabled: "true"
            detect_args: --detect.blackduck.rapid.compare.mode=ALL`

I am passing my project version using `DETECT_PROJECT_VERSION_NAME` environment variable

**Comment by guellaeq** [comment]
@PiotrMorawiec-TomTom Thanks for pointing it out. It seems the link is not available. Is your idea publicly available yet ?

**Comment by PiotrMorawiec-TomTom** [comment]
@guellaeq It appears there isn’t an option to set the visibility to public or private. You may need to log in with your community account to access this.
